model/src/main/kotlin/config/PackageConfiguration.kt

```diff
@@ -66,25 +68,55 @@ data class PackageConfiguration(
      * License finding curations.
      */
     @JsonInclude(JsonInclude.Include.NON_EMPTY)
-    val licenseFindingCurations: List<LicenseFindingCuration> = emptyList()
+    val licenseFindingCurations: List<LicenseFindingCuration> = emptyList(),
+
+    /**
+     * The source code origin this configuration applies to.
+     */
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    val sourceCodeOrigin: SourceCodeOrigin? = null
 ) {
     init {
-        require((sourceArtifactUrl == null) xor (vcs == null)) {
-            "A package configuration must either set the 'sourceArtifactUrl' or the 'vcs' property."
+        val vcsOrSourceArtifact = (sourceArtifactUrl == null) xor (vcs == null)
+        require(
+            vcsOrSourceArtifact || (sourceCodeOrigin != null && vcs == null) ||
+                (sourceCodeOrigin == null && vcs == null && sourceArtifactUrl == null)
+        ) {
+            "A package configuration must either set the 'sourceArtifactUrl' or the 'vcs', or the 'sourceCodeOrigin' " +
```

Comment:
Consider saying: `A package configuration must not set more than one of 'sourceArtifactUrl', 'vcs' or 'sourceCodeOrigin' or

`A package configuration must set at most one of 'sourceArtifactUrl', 'vcs' or 'sourceCodeOrigin'

Reply:
Fixed in https://github.com/oss-review-toolkit/ort/pull/10377.